### DISCUSSION TITLE
VS Code needs

### DISCUSSION THREAD

**Discussion by agracey** [epinio/epinio#1144] [Ideas]
Discussion about what we can do to improve the integration with VS Code

**Comment by gunamata** [comment]
Here's the VS Code extension work in progress repository.
https://github.com/rancher-sandbox/vscode-plugin-epinio.git 

Note: It's been a while I last tested the extension. I will spend some time in the coming days to test the extension with the most recent epinio version and make necessary adjustments. Plus, the api calls implementation needs to be done for app push/repush functionality.

**Reply by agracey** [reply]
Thank you!

**Comment by agracey** [comment]
From discussion with Dimitris and Olivier:

The first set of functionality that we want to target is a user pressing F5 and the app being pushed (via the API, without calling the epinio cli). When a push succeeds, we should show a toast message and a link to the app's ingress URL. 

Stretch goals could be:

- Watch code for changes and repush after x seconds (configurable)
- Give a deep link to the app in the epinio UI
- Shift-F5 to also trigger a port forward after push

**Reply by agracey** [reply]
As part of this work, we also need to refactor the work that Guna did to use the API. We should likely generate a client library from the openapi spec and call into that.
